snapcraft/internal/sources/_7z.py

```diff
@@ -0,0 +1,62 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Tim Süberkrüb
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import libarchive
+import shutil
+import tempfile
+
+from . import errors
+from ._base import FileBase
+
+
+class _7z(FileBase):
```

Comment:
> it will just be a little confusing with our understanding of _ as internal.

Yeah... thinking on that a little more, I change my vote from neutral to "please spell it out."